index.html

```diff
@@ -536,7 +537,9 @@ <h2><a href="#custom-canvas" id="custom-canvas" class="anchor">Custom Canvas</a>
     if (window.matchMedia) {
       var mediaQuery = window.matchMedia('(prefers-color-scheme: light)');
 
-      if (mediaQuery.matches) {
+      if (lastSavedTheme && lastSavedTheme === 'light') {
```

Comment:
I feel like balancing a last saved theme with user-agent preferences would get difficult and confusing, especially on operating systems that automatically change between light and dark themes based on the time of day. Trying to follow the logic of what exactly happens in each case and I am already confused. It feels like there should really be a 3-way switch between light/dark/auto, and only when the user has the theme on auto would it automatically change based on user-agent preferences. What do you think?

Reply:
I agree. The previous implementation can sometimes be unpredictable. I thought it too, but I didn't want to overkill this functionality, so I was waiting for your feedback to see if I was the only one noticing it.